devito/ir/iet/visitors.py

```diff
@@ -233,6 +135,9 @@ def _args_decl(self, args):
         """Convert an iterable of :class:`Argument` into cgen format."""
         ret = []
         for i in args:
+            if not hasattr(i, "is_ScalarArgument"):
+                from devito.arguments import ArgumentVisitor
```

Comment:
do we have a cyclic dependence if we move this to the top?

Reply:
It does indeed. This snippet can probably not go in master in its current form anyway as this is a very hard dependence on the arguments module - seems very wrong. Help me figure out a way to avoid doing this at all?

Reply:
we might have to split code generation into multiple steps, such as
* collection of arguments
* generation of body (CGen)
* generation of arguments
* etc ?

Not sure if this actually makes sense